File: rpncalc/sample_solutions/python/calc.py
import operator

operators = {'+': operator.add, '-': operator.sub, '*': operator.mul, '/': operator.floordiv}

while True:
    stack = []
    tokens = input().split()

    if tokens[0] == '#':
        break

    for token in tokens:
        try:
            stack.append(int(token))
        except ValueError:
            if token in operators:
                b = stack.pop()
                a = stack.pop()
                stack.append(operators[token](a, b))
            else:
                print('Invalid number or operator')

    print(stack.pop())

File: rpncalc/sample_solutions/java/Calc.java
import java.util.*;
import java.util.function.BiFunction;

class Calc {
    static Map<String, BiFunction<Integer, Integer, Integer>> operators = new HashMap<>() {{
        put("+", (a, b) -> a + b);
        put("-", (a, b) -> a - b);
        put("*", (a, b) -> a * b);
        put("/", (a, b) -> a / b);
    }};

    private static void eval(String op, Deque<Integer> stack) {
        Integer b = stack.pop();
        Integer a = stack.pop();
        stack.push(operators.get(op).apply(a, b));
    }

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        Deque<Integer> stack = new ArrayDeque<>(512);

        while (true) {
            String line = s.nextLine();
            if (line.equals("#"))
                break;

            String[] components = line.split("\\s");
            for (String component : components) {
                try {
                    stack.push(Integer.parseInt(component));
                } catch (NumberFormatException e) {
                    if (operators.containsKey(component)) {
                        eval(component, stack);
                    } else {
                        System.out.println("Invalid number or operator");
                    }
                }
            }
            System.out.println(stack.pop());
        }
    }
}